plugins/modules/system/kernel_blacklist.py

```diff
@@ -44,109 +46,64 @@
 import os
 import re
 
-from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.module_helper import StateModuleHelper
 
 
-class Blacklist(object):
-    def __init__(self, module, filename, checkmode):
-        self.filename = filename
-        self.module = module
-        self.checkmode = checkmode
+class Blacklist(StateModuleHelper):
+    output_params = ('name', 'state')
+    module = dict(
+        argument_spec=dict(
+            name=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['absent', 'present']),
+            blacklist_file=dict(type='str', default='/etc/modprobe.d/blacklist-ansible.conf'),
+        ),
+        supports_check_mode=True,
+    )
 
-    def create_file(self):
-        if not self.checkmode and not os.path.exists(self.filename):
-            open(self.filename, 'a').close()
-            return True
-        elif self.checkmode and not os.path.exists(self.filename):
-            self.filename = os.devnull
-            return True
+    def __init_module__(self):
+        self.vars.filename = self.default_file if not self.vars.blacklist_file else self.vars.blacklist_file
+        self.vars.set('file_exists', os.path.exists(self.vars.filename), output=False, change=True)
+        if not self.vars.file_exists:
+            open(self.vars.filename, 'a').close()
```

Comment:
I think it's better to always use `with` with `open()` when possible:
```suggestion
            with open(self.vars.filename, 'a'):
                pass
```